File: frontend/services/claim.service.ts
import { fetchWithAuth } from "../utils/api";

export const getClaims = async () => {
  const response = await fetchWithAuth("/claims");
  if (!response.ok) throw new Error("Gagal mengambil daftar klaim");
  return response.json();
};

export const createClaim = async (data: { claimerName: string; claimerPhone: string; reportId: number }) => {
  const response = await fetchWithAuth("/claims", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  });

  if (!response.ok) throw new Error("Gagal mendaftarkan klaim");
  return response.json();
};

export const updateClaimStatus = async (id: number, status: string) => {
  const response = await fetchWithAuth(`/claims/${id}/status`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ status }),
  });

  if (!response.ok) throw new Error("Gagal memperbarui status klaim");
  return response.json();
};

File: mobile/lib/services/claim_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:flutter_dotenv/flutter_dotenv.dart';

class ClaimService {

  static String get baseUrl =>
      dotenv.env['BASE_URL']!;

  static Future<List<dynamic>>
      getClaims() async {

    final response =
        await http.get(
      Uri.parse(
        "$baseUrl/claims",
      ),
    );

    if (response.statusCode !=
        200) {
      throw Exception(
        "Gagal mengambil claim",
      );
    }

    return jsonDecode(
      response.body,
    );
  }

  static Future<dynamic>
      createClaim(
    String claimerName,
    String claimerPhone,
    int reportId,
  ) async {

    final response =
        await http.post(
      Uri.parse(
        "$baseUrl/claims",
      ),
      headers: {
        "Content-Type":
            "application/json",
      },
      body: jsonEncode({
        "claimerName":
            claimerName,
        "claimerPhone":
            claimerPhone,
        "reportId":
            reportId,
      }),
    );

    if (response.statusCode !=
            201 &&
        response.statusCode !=
            200) {
      throw Exception(
        "Gagal mengajukan claim",
      );
    }

    return jsonDecode(
      response.body,
    );
  }
}